examples/discovery/server_register.c

```diff
@@ -80,55 +94,73 @@ int main(int argc, char **argv) {
     //UA_String caps = UA_String_fromChars("LDS");
     //config.serverCapabilities = &caps;
 
-    /* add a variable node to the address space */
+    /* Create a data source variable node with read/write callbacks */
     UA_Int32 myInteger = 42;
     UA_NodeId myIntegerNodeId = UA_NODEID_STRING(1, "the.answer");
     UA_QualifiedName myIntegerName = UA_QUALIFIEDNAME(1, "the answer");
-    UA_DataSource dateDataSource;
-    dateDataSource.read = readInteger;
-    dateDataSource.write = writeInteger;
+    
+    /* Configure data source callbacks */
+    UA_DataSource integerDataSource;
+    integerDataSource.read = readInteger;
+    integerDataSource.write = writeInteger;
+    
+    /* Set variable attributes */
     UA_VariableAttributes attr = UA_VariableAttributes_default;
     attr.description = UA_LOCALIZEDTEXT("en-US", "the answer");
     attr.displayName = UA_LOCALIZEDTEXT("en-US", "the answer");
 
-    UA_Server_addDataSourceVariableNode(server, myIntegerNodeId,
+    /* Add the variable node to the address space */
+    UA_StatusCode retval = UA_Server_addDataSourceVariableNode(server, myIntegerNodeId,
                                         UA_NS0ID(OBJECTSFOLDER), UA_NS0ID(ORGANIZES),
-                                        myIntegerName, UA_NODEID_NULL, attr, dateDataSource,
+                                        myIntegerName, UA_NODEID_NULL, attr, integerDataSource,
                                         &myInteger, NULL);
+    if(retval != UA_STATUSCODE_GOOD) {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER,
+                     "Could not add variable node. StatusCode %s",
+                     UA_StatusCode_name(retval));
+        goto cleanup;
+    }
 
     UA_Server_run_startup(server);
 
-    // register server
+    /* Configure client for discovery server communication */
     UA_ClientConfig cc;
     memset(&cc, 0, sizeof(UA_ClientConfig));
     UA_ClientConfig_setDefault(&cc);
 
-    UA_StatusCode retval =
-        UA_Server_registerDiscovery(server, &cc,
+    /* Register server with the discovery server */
+    retval = UA_Server_registerDiscovery(server, &cc,
                                     UA_STRING(DISCOVERY_SERVER_ENDPOINT), UA_STRING_NULL);
     if(retval != UA_STATUSCODE_GOOD) {
         UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER,
                      "Could not create periodic job for server register. StatusCode %s",
                      UA_StatusCode_name(retval));
-        UA_Server_delete(server);
-        return EXIT_FAILURE;
+        UA_ClientConfig_clear(&cc);
+        goto cleanup;
     }
 
+    /* Main server loop - process client requests */
     while(running)
         UA_Server_run_iterate(server, true);
 
-    // Unregister the server from the discovery server.
-    memset(&cc, 0, sizeof(UA_ClientConfig));
-    UA_ClientConfig_setDefault(&cc);
```

Comment:
You need to recreate the client config.
It gets cleared internally in the _register method.

This also impacts the cleanup below.